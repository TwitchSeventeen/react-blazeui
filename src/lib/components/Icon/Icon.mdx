---
name: Icon
menu: Components
---

import { Playground, PropsTable } from 'docz'
import Icon from './index';

# Icon

The Icon component is used to display an icon. The complete (free) version of Font Awesome 5.6.3 has been included.

**NOTE:** The Icon component is not part of Blaze UI.

## Basic usage
<Playground>
    <Icon name="calendar-alt" />
</Playground>

## Size
Use the **size** prop to set the Icon size.
<Playground>
    <Icon name="calendar-alt" size="xs" /> xs <br />
    <Icon name="calendar-alt" size="sm" /> sm <br />
    <Icon name="calendar-alt" size="lg" /> lg <br />
    <Icon name="calendar-alt" size="2x" /> 2x <br />
    <Icon name="calendar-alt" size="3x" /> 3x <br />
    <Icon name="calendar-alt" size="4x" /> 4x <br />
    <Icon name="calendar-alt" size="5x" /> 5x <br />
    <Icon name="calendar-alt" size="6x" /> 6x <br />
    <Icon name="calendar-alt" size="7x" /> 7x <br />
    <Icon name="calendar-alt" size="8x" /> 8x <br />
    <Icon name="calendar-alt" size="9x" /> 9x <br />
    <Icon name="calendar-alt" size="10x" /> 10x <br />
</Playground>

## Flip
Flip the Icon using the **flipped** prop
<Playground>
    <Icon name="snowman" flipped="vertical" size="3x" />
</Playground>

## Rotate
Rotate the Icon using the **rotate** prop
<Playground>
    <Icon name="snowman" rotate={90} size="3x" />
    <Icon name="snowman" rotate={180} size="3x" />
    <Icon name="snowman" rotate={270} size="3x" />
</Playground>

## Spin
Spin the Icon using the **spin** prop.

Use the **pulse** prop to spin the Icon with 8 steps.
<Playground>
    <Icon name="spinner" spin size="3x" />
    <br />
    <Icon name="spinner" pulse size="3x" />
</Playground>

## Type
The **type** prop specifies the style of icon. By default it's _'fas'_ (Solid). 
<Playground>
    <Icon name="spinner" size="3x" />
    <br />
    <Icon name="android" type="fab" size="3x" />
</Playground>

## Properties
<PropsTable of={Icon} />