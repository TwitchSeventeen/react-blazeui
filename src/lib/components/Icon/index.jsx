import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Icon (props) {
  const { className, name, type, size, flipped, rotate, spin, pulse } = props;
  const classes = classNames(
      'c-icon',
      { [`fa-${size}`]: size },
      { [`fa-flip-${flipped}`]: flipped },
      { [`fa-rotate-${rotate}`]: rotate },
      { [`fa-${name}`]: name },
      { [type]: type },
      { 'fa-spin': spin },
      { 'fa-pulse': pulse },
      className
  );

  return (
      <i {...props} className={classes} />
  );
}

Icon.propTypes = {
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Name of the icon
   */
  name: PropTypes.string.isRequired,
  /**
   * Icon set to use
   */
  type: PropTypes.string,
  /**
   * Flips the icon
   */
  flipped: PropTypes.oneOf([
      'horizontal',
      'vertical',
  ]),
  /**
   * Rotates the icon
   */
  rotate: PropTypes.oneOf([
      90,
      180,
      270,
  ]),
  /**
   * Icon set to use
   */
  size: PropTypes.oneOf([
      'xs',
      'sm',
      'lg',
      '2x',
      '3x',
      '4x',
      '5x',
      '6x',
      '7x',
      '8x',
      '9x',
      '10x',
  ]),
  /**
   * Spins the Icon
   */
  spin: PropTypes.bool,
  /**
   * Spins the Icon with 8 steps
   */
  pulse: PropTypes.bool,
};

Icon.defaultProps = {
  className: null,
  size: null,
  type: 'fas',
  flipped: null,
  rotate: null,
  spin: false,
  pulse: false,
};

export default Icon;