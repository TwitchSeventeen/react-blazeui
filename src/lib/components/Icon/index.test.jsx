import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Icon from './index';

describe('Icon', () => {
  it('should be defined', () => {
    expect(Icon).toBeDefined();
  });

  it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <Icon name="spinner" />,
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
