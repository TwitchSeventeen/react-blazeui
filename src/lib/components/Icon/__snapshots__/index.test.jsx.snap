// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<i
  aria-hidden={true}
  className="c-icon fa-spinner fas"
/>
`;
